presto-parquet/src/main/java/com/facebook/presto/parquet/reader/ParquetReader.java

```diff
@@ -466,6 +468,94 @@ public Block readBlock(Field field)
         return readColumnChunk(field).getBlock();
     }
 
+    int getLevel(Type rootType, Type leafType)
+    {
+        int level = 0;
+        Type currentType = rootType;
+        while (!currentType.equals(leafType)) {
```

Comment:
I am just concerned we run into an infinite loop in some corner cases. Normally we shouldn't but if that happens, it would be hard to debug. At least to add debug log to it in case it happens, debug log can tell. 

It seems the infinite loop can be avoided with an assumption that leafType is always equal to one of the types on the paths while navigating. But this method is called by readPushedDownSubfieldBlock() which is a public method. And the types are passed in parameters. So we need to add some check or at least at debug log to prepare for it.